src/gui/Layout.cpp

```diff
@@ -295,103 +276,85 @@ void Layout::layoutPages()
 							paddingRight = XOURNAL_ROOM_FOR_SHADOW;
 						}
 						else
-						{								//align left
+						{                                //align left
 							paddingLeft = XOURNAL_ROOM_FOR_SHADOW;
 							paddingRight = XOURNAL_PADDING_BETWEEN - XOURNAL_ROOM_FOR_SHADOW + columnPadding;
 						}
 					}
 					else
-					{	// not paired page mode - center
+					{    // not paired page mode - center
 						paddingLeft = XOURNAL_PADDING_BETWEEN / 2 + columnPadding / 2;      //center justify
 						paddingRight = XOURNAL_PADDING_BETWEEN - paddingLeft + columnPadding / 2;
 					}
 
 					x += paddingLeft;
 
-					v->setX(x);		//set the page position
+					v->setX(x);        //set the page position
 					v->setY(y);
 
 					x += vDisplayWidth + paddingRight;
-
 				}
 			}
 			else
 			{
-				x += this->sizeCol[c] + XOURNAL_PADDING_BETWEEN;
+				x += this->widthCols[c] + XOURNAL_PADDING_BETWEEN;
 			}
 		}
 		x = borderX;
-		y += this->sizeRow[r] + XOURNAL_PADDING_BETWEEN;
+		y += this->heightRows[r] + XOURNAL_PADDING_BETWEEN;
 
 	}
 
 
+	//This seems to be wrong:
 	int totalWidth = borderX;
-	for (int c = 0; c < this->columns; c++)
+	for (int c = 0; c < columns; c++)
 	{
-		totalWidth += this->sizeCol[c] + XOURNAL_PADDING_BETWEEN;
-		this->sizeCol[c] = totalWidth;	//accumulated - absolute pixel location for use by getViewAt() and updateVisibility()
+		totalWidth += this->widthCols[c] + XOURNAL_PADDING_BETWEEN;
+		this->widthCols[c] = totalWidth;    //accumulated - absolute pixel location for use by getViewAt() and updateVisibility()
 	}
-	totalWidth += borderX - XOURNAL_PADDING_BETWEEN;
 
-	
+	//This seems to be wrong:
```

Comment:
Is this fixed now or is this a TODO?

Reply:
Oh right, this is a Todo. This code snipped is rather wrong but it definitely lacks of understanding. We may want to extract it / reformat it. It also is the reason, that we musst recalculate the layout every time "layoutPages" is called. It invalidates the precalculation.
I had to introduce a new variable to catch this.

```suggestion
```